solutions/cf-1365D.mdx

```diff
@@ -40,97 +40,105 @@ typedef long long ll;
+		return;
+	visited[r][c] = true;
+	floodfill(r + 1, c);
+	floodfill(r - 1, c);
+	floodfill(r, c + 1);
+	floodfill(r, c - 1);
 }
 
-void solve() {
-    memset(grid, '.', sizeof(grid));
-    memset(visited, 0, sizeof(visited));
-    cin >> N >> M;
-    for(int i = 0; i < N; i++) {
-        for(int j = 0; j < M; j++) {
-            cin >> grid[i][j];
-        }
-    }
-    //surrounds the bad people
-    bool ok = 1;
-    for(int i = 0; i < N; i++) {
-        for(int j = 0; j < M; j++) {
-            if(grid[i][j] == 'B') {
-                if(i - 1 >= 0) {
-                    if(grid[i - 1][j] == 'G') {
-                        ok = 0;
-                    }
-                    if(grid[i - 1][j] == '.') {
-                        grid[i - 1][j] = '#';
-                    }
-                }
-                if(i + 1 < N) {
-                    if(grid[i + 1][j] == 'G') {
-                        ok = 0;
-                    }
-                    if(grid[i + 1][j] == '.') {
-                        grid[i + 1][j] = '#';
-                    }
-                }
-                if(j - 1 >= 0) {
-                    if(grid[i][j - 1] == 'G') {
-                        ok = 0;
-                    }
-                    if(grid[i][j - 1] == '.') {
-                        grid[i][j - 1] = '#';
-                    }
-                }
-                if(j + 1 < M) {
-                    if(grid[i][j + 1] == 'G') {
-                        ok = 0;
-                    }
-                    if(grid[i][j + 1] == '.') {
-                        grid[i][j + 1] = '#';
-                    }
-                }
-            }
-        }
-    }
-    if(!ok) {
-        cout << "No" << "\n";
-        return;
-    }
-    ok = 1;
-    floodfill(N - 1, M - 1);
-    for(int i = 0; i < N; i++) {
-        for(int j = 0; j < M; j++) {
-            if(grid[i][j] == 'G' && !visited[i][j]) {
-                ok = 0;
-                break;
-            }
-        }
-    }
-    if(!ok) {
-        cout << "No" << "\n";
-        return;
-    }
-    cout << "Yes" << "\n";
+void solve()
+{
+	memset(grid, '.', sizeof(grid));
+	memset(visited, 0, sizeof(visited));
+	cin >> N >> M;
+	for (int i = 0; i < N; i++)
+	{
+		for (int j = 0; j < M; j++)
+		{
+			cin >> grid[i][j];
+		}
+	}
+	//surrounds the bad people
+	bool ok = 1;
+	for (int i = 0; i < N; i++)
+	{
+		for (int j = 0; j < M; j++)
+		{
+			if (grid[i][j] == 'B')
+			{
+				for (int x = 0; x < 4; x++)
+				{ // the 4 directions
+					int newRow = i + rowMovement[x];
+					int newColumn = j + columnMovement[x];
+					// check if it is in the boundaries
+					if (newRow >= 0 && newRow < N && newColumn >= 0 && newColumn < M)
+					{
+						if (grid[newRow][newColumn] == 'G')
+						{ // make false
+							ok = false;
+						}
+						if (grid[newRow][newColumn] == '.')
+						{ // turn it into #
+							grid[newRow][newColumn] = '#';
+						}
+					}
+				}
+			}
+		}
+	}
+	if (!ok)
+	{
+		cout << "No"
+			 << "\n";
```

Comment:
what's with the indentation here

Reply:
not my code, only added modifications

Reply:
it's not like it changes the structure lol